### DISCUSSION TITLE
Error 403 - Service accounts cannot invite attendees without Domain-Wide Delegation of Authority where add Attendee

### DISCUSSION THREAD

**Discussion by kobazzo** [spatie/laravel-google-calendar#210] [Q&A]
I have done all steps of the documentation, but if i create ad event with Attendee there is ad error:

"Error 403 - Service accounts cannot invite attendees without Domain-Wide Delegation of Authority"

**Comment by mrabbani** [comment]
Did you solve this issue @kobazzo ?

**Reply by fuhrn** [reply]
1) be sure you have a G Suite Workspace account (does not work with common gmail account calendar)

2) in Google Cloud Platform -> IAM & Admin -> Service Account -> Show Domain Wide Delegation. Check 'Enable Google Workspace Domain-wide Delegation' and include your service account Client ID (all numbers). You can get this number from you service_account_credential.json file.

3) In GoogleCalendarFactory.php,  insert in 'protected static function createServiceAccountClient()'    
            $client->setSubject('user to impersonate'); e.g. 'johndoe@example.com', check google API docs for more details. 
This is the account that is the owner or super administrator of the project in google workspace, that is using Calendar API. 
Create an .env variable for this account and include it in google_calendar.php config array. Use this config variable in 'user to impersonate'.

4) when testing, be sure to include $optParams = ['sendNotifications' => true] as a parameter of method 'save'

I tested sending invitations to accounts on gmail, yahoo and Hotmail. All works ok, receiving invitations, and when you get responds, event in calendar reflects approval (or not). Also works great if you cancel event.

example:

use \Spatie\GoogleCalendar\Event;

   $event = new Event();

    $event->name = 'por fin funciona LPM';
    $event->startDateTime = \Carbon\Carbon::now();
    $event->endDateTime = \Carbon\Carbon::now()->addHour();
    $event->addAttendee([
        'email' => 'juan.perez@example.com',
        'name' => 'juan',
        'comment' => 'prueba de la API',
    ]);

    $optParams = [
        'sendNotifications' => true
    ];

    $event->save('insertEvent', $optParams);

    dd($event);

**Reply by 82solutions** [reply]
"Error 403 - Service accounts cannot invite attendees without Domain-Wide Delegation of Authority"

**Comment by wasitmirani** [comment]
I have done all steps of the documentation but i'm also phasing issues
"Error 403 - Service accounts cannot invite attendees without Domain-Wide Delegation of Authority"

**Comment by WaheedHussainHaans** [comment]
Any update on this please? TIA

**Comment by mikesosa** [comment]
Any update on this??

**Comment by troutman21** [comment]
Also having the same issue here. Anyone figure this piece out?

**Comment by yanuadin** [comment]
It works. In the step 3 just add variable GOOGLE_CALENDAR_IMPERSONATE="" on your .env. Thanks
